### DISCUSSION TITLE
Is there a way to support external authentication (OIDC) with KeyGen.sh ?

### DISCUSSION THREAD

**Discussion by chintanp** [keygen-sh/community#130] [Q&A] [answered]
We have a dedicated Keycloak (https://www.keycloak.org/) server for authentication for IAM to all our web services. We would like to keep that intact and use the same token to authenticate into Keygen.sh. If yes, is this documented anywhere? And further, would it be possible to use the Keycloak provided roles in Keygen?

**Comment by ezekg** [comment, marked as answer]
It's on the roadmap, but not currently being worked on yet. See https://github.com/keygen-sh/keygen-api/issues/409.
